In [76]:
from math import *

In [1]:
numero = 9.33
frase = "Hola"
complejo = 3+5J
mi_lista = [frase, 4, 3.55, +4+1J ]
mi_tupla = (frase, 4, 3.55, +4+1J )
numero/2 , numero + 2, frase[0], mi_lista[0][1]

(4.665, 11.33, 'H', 'o')


### Función que calcula es factorial de un número:

In [34]:
def mi_factorial(n):
    
    fac = 1
    for i in range(n):
        fac *= i+1
    
    return fac

In [39]:
mi_factorial(19)

121645100408832000

 * "Al trabajar con una computadora debemos considerar las características de la misma (hardware/software)..."

In [58]:
def mi_seno(x, k):   # k es el orden de precisión de la serie de Taylor
    seno = 0
    for n in range(k):
        seno = seno + (-1)**n * (x**(2*n+1)/ mi_factorial(2*n+1) )
    
    return seno

In [59]:
mi_seno(1, 5)

0.8414710097001764

In [79]:
mi_seno(pi/2, 5)

1.0000035425842861

In [80]:
mi_seno(pi/2, 15)

1.0000000000000002

In [85]:
mi_seno(pi/2, 50)

1.0000000000000002

Con "math" :

In [77]:
sin(1)

0.8414709848078965

In [81]:
sin(pi/2)

1.0

________________

#### ¿Cuál es el número más grande que puedo escribir en 64 bits?

**Respuesta**

Sin son 64 bits, encontramos 64 posiciones de 0 ó 1, donde la computadora ocupa un bit (primera posición) para almacenar el signo. Por lo tanto, el máximo número que se podría almacenar sería de $ 1x2^{63} $.

In [71]:
number = 0
for i in range(63):
    number += 2**i
    
print(number)

9223372036854775807


En 64 bits tendríamos un rango de $ -9223372036854775807 $ a $ 9223372036854775807 $


En 32 bits tendríamos :

In [70]:
number = 0
for i in range(31):
    number += 2**i
    
print(number)

2147483647
